lib/compat/wordpress-7.0/class-gutenberg-rest-templates-controller.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+class Gutenberg_REST_Templates_Controller_7_0 extends WP_REST_Templates_Controller {
+	/**
+	 * Prepare a single template output for response
+	 *
+	 * @since 5.8.0
+	 * @since 5.9.0 Renamed `$template` to `$item` to match parent class for PHP 8 named parameter support.
+	 * @since 6.3.0 Added `modified` property to the response.
+	 *
+	 * @param WP_Block_Template $item    Template instance.
+	 * @param WP_REST_Request   $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+
+		error_log( '!!prepare_item_for_response %%%%%%%%%%%' );
+		// Don't prepare the response body for HEAD requests.
+		if ( $request->is_method( 'HEAD' ) ) {
+			return new WP_REST_Response( array() );
+		}
+
+		/*
+		 * Resolve pattern blocks so they don't need to be resolved client-side
+		 * in the editor, improving performance.
+		 */
+		$blocks        = parse_blocks( $item->content );
+		$blocks        = gutenberg_resolve_pattern_blocks( $blocks );
```

Comment:
This is the only change.

Reply:
Should we add a code comment indicating that so it's clear at a glance that it's the only diff (and doesn't need a backport)?

Reply:
Okay, after a few 🤦🏻 🤦🏻 I see I can use `get_block_templates`

Update incoming...

Reply:
> Can't we use a filter so it applies to all endpoints?

Back to this. No we can't. 😄 

Caused this: https://core.trac.wordpress.org/ticket/64279

The TL;DR is that, by parsing the patterns again the render order is upset and it upsets short codes.